datafusion-cli/src/main.rs

```diff
@@ -177,13 +177,14 @@ async fn main_inner() -> Result<()> {
 
     // set disk limit
     if let Some(disk_limit) = args.disk_limit {
-        let disk_manager = DiskManager::try_new(DiskManagerConfig::NewOs)?;
+        let mut disk_manager = DiskManager::try_new(DiskManagerConfig::NewOs)?;
```

Comment:
What would be nicer would be if try_new didn't return an Arc -- but that is a more disruptive change so I think this solution improves things even if it is non ideal